### DISCUSSION TITLE
Method override completion

### DISCUSSION THREAD

**Discussion by haringsrob** [phpactor/phpactor#2442] [Ideas]
I do not always know for certain what the method is I want to override,

```
class X extends B {
  function <CUR>
}
```

I think it would be useful to allow override autocomplete based on the cursor position in the example.

**Comment by dantleech** [comment]
yes, this is a long overdue feature, the functionality exists and is exposed by the legacy RPC but it's not exposed by the language server integration.

the only question is if it should be an auto-complete suggestion or a code action (override method)

**Reply by haringsrob** [reply]
Not sure how it would work for for an action, in phpstorm it is quite convenient as autocomplete.

![Screenshot 2023-11-23 at 12 26 17](https://github.com/phpactor/phpactor/assets/866743/7b943c17-0ff8-4945-8bbd-25ef0ce695f3)
